lib/assets/gql-generator.js

```diff
@@ -281,6 +281,16 @@ module.exports = {
 
       /* Union types */
       if (curType.astNode && curType.astNode.kind === 'UnionTypeDefinition') {
+
+        /* Make sure UnionTypeDefinition are also not circularly referenced */
+        if ((crossReferenceKeyList.hasOwnProperty(crossReferenceKey) && crossReferenceKeyList[crossReferenceKey]) ||
+          curDepth > depthLimit
+        ) {
+          crossReferenceKeyList[crossReferenceKey] = false;
+          return '';
```

Comment:
@VShingala Will there be a case where a self referencing union type will *not* be skipped?

Reply:
@webholik For first time, it will not be skipped. But once visited, corresponding key will be marked under `crossReferenceKeyList`. And upon further reference it'll be skipped.

Reply:
@VShingala If a reference is cyclic, then no matter the depth limit it will always be skipped, right? Do we then need to check the depth limit?

Reply:
@webholik Yes, for other cases where it's not cyclic, we still need to make sure not resolve past depth limit. This logic was not present already for UnionTypes, which is why existing test case was failing where more levels were getting resolved than default 1.

https://github.com/postmanlabs/graphql-to-postman/pull/57#discussion_r1552861673